fleet-be/src/main/java/com/everis/training/fleet/business/fleet/boundary/VehicleResource.java

```diff
@@ -30,28 +30,44 @@ public Response listVehicles() {
     return Response.ok().entity(con.getAllVehicles()).build();
   }
 
-  @Path("free")
+  @Path("free-vehicles")
   @GET
   public Response listFreeVehicles() {
     return Response.ok().entity(con.getAllFreeVehicles()).build();
   }
 
+  @Path("{customerId}/reserve-vehicle/{vin}")
   @POST
-  public Response addVehicle(Vehicle vehicle) {
-    con.add(vehicle);
-    return Response.ok().entity(String.format("The vehicle has been added successfully!")).build();
+  public Response reserveCar(@PathParam("customerId") final Integer id, @PathParam("vin") String vin){
+    String response = "The reserve has been done successfully!";
+    try {
+      con.reserveCar(id, vin);
+    } catch (Exception e) {
+      response=e.getMessage();
+    }
+    return Response.ok().entity(response).build();
   }
 
-  @PUT
-  public Response updateVehicle(Vehicle vehicle) {
-    con.update(vehicle);
-    return Response.ok().entity(String.format("The vehicle has been updated successfully!")).build();
+  @POST
+  public Response addVehicle(Vehicle vehicle) {
+    String response = "The vehicle has been added successfully!";
+    try {
+      con.add(vehicle);
+    } catch (Exception e) {
+      response=e.getMessage();
```

Comment:
We will see a better way to report the exceptions with REST. At the momement, keep it as is.